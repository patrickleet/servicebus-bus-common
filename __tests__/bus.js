const bus = require('bus')

const {
  makeBus,
  handleError,
  throwErr
} = bus

jest.mock('servicebus', () => {
  return {
    bus: () => {
      return {
        use: jest.fn(),
        logger: jest.fn(),
        package: jest.fn(),
        correlate: jest.fn()
      }
    }
  }
})
jest.mock('servicebus-retry')

describe('lib/bus', () => {
  it('should make a bus when makeBus is called', () => {
    expect(handleError).toBeDefined()
    expect(makeBus).toBeDefined()
    expect(throwErr).toBeDefined()

    let bus = makeBus()
    expect(bus).toMatchSnapshot()
  })

  it('should throw an error when throwErr is called', () => {
    expect(()=>{
      throwErr(new Error('OMG'))
    }).toThrow() 
  })

  it('should handle an error with handleError is called', () => {
    expect(()=>{
      handleError('msg', 'err')
    }).toThrow()
  })

})
