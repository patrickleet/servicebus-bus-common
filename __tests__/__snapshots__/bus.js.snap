// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/bus should make a bus when makeBus is called 1`] = `
Object {
  "correlate": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "logger": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "package": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
  "use": [MockFunction] {
    "calls": Array [
      Array [
        undefined,
      ],
      Array [
        undefined,
      ],
      Array [
        undefined,
      ],
      Array [
        undefined,
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
      Object {
        "isThrow": false,
        "value": undefined,
      },
      Object {
        "isThrow": false,
        "value": undefined,
      },
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;
